src/Dapr.Workflow/WorkflowServiceCollectionExtensions.cs

```diff
@@ -29,18 +30,29 @@ public static class WorkflowServiceCollectionExtensions
     /// </summary>
     /// <param name="serviceCollection">The <see cref="IServiceCollection"/>.</param>
     /// <param name="configure">A delegate used to configure actor options and register workflow functions.</param>
+    /// <param name="configureDaprClient">A delegate used to optionally configure the Dapr client connection.</param>
     /// <param name="lifetime">The lifetime of the registered services.</param>
     public static IServiceCollection AddDaprWorkflow(
         this IServiceCollection serviceCollection,
         Action<WorkflowRuntimeOptions> configure,
+        Action<IServiceProvider, DaprClientBuilder>? configureDaprClient = null,
         ServiceLifetime lifetime = ServiceLifetime.Singleton)
     {
         if (serviceCollection == null)
         {
             throw new ArgumentNullException(nameof(serviceCollection));
         }
 
-        serviceCollection.AddDaprClient(lifetime: lifetime);
+
+        if (configureDaprClient is not null)
```

Comment:
I'd like to be able to have users install the various packages potentially independently of one another. If I want to connect to my local Dapr runtime for pubsub, but connect to a remote Dapr runtime for state management, that wouldn't be trivially done by having the developer call `AddDaprClient()` as it would supersede the per-client options.

Perhaps in the shorter term then, instead of what I've got here, I need to refactor it all to output a `DaprConnectionOptions` or the like and within each of the clients, stand up an unregistered instance of `DaprClient` using those options. 

Then, if you want the `DaprClient` (as you would for most of today's Dapr functionality), you use `AddDaprClient()` and if you want any of the other clients, you use `AddWhateverClient()`, but the underlying `DaprClient`s themselves are discrete from one another.

In the short term, it ensures you don't have lifecycle conflicts across the board. In the really long term, it means we can eventually drop `AddDaprClient()` altogether once all the existing functionality is migrated into the purpose-built clients.

What do you think?

Reply:
I was thinking a little more about this - today, because all the clients use static builders, it's insufficient to rely on the previously created DaprClient instances to get the configuration as the builders are recreating the connections from scratch.

So while the configuration bits may be available in the `IConfiguration`, that's not necessarily going to cover all configuration scenarios (e.g. setting their own timeout, specifying their own gRPC options or their own JSON serialization settings), it'd at least handle setting endpoints (HTTP and gRPC) and the API token.

Reply:
@philliphoff I'm thinking that it'd be easier to just pull the various values from an `IConfiguration` rather than pass in all manner of optional actions for each of the dependency client builders. That way, we get that benefit of setting the configuration values in one place in each app and using it for each of the various Dapr clients and if developers really need to use an alternative configuration, they can use the respective static builders instead and override the values.

I've implemented this for Jobs at https://github.com/dapr/dotnet-sdk/pull/1417 in a manner similar to one of the files changed in this PR, but with documentation updates and a little more testing.